File: Beginner/2_Count_Vowels/my_solution/cpp/count_vowels.cpp
#include "count_vowels.hpp"
// Function definition
int countVowels(std::string string_to_count)
{
    int count = 0;
    std::string vowels = "aeiou";
    for(int i = 0; i < string_to_count.length(); i++)
    {
        if(vowels.find(string_to_count[i]) != std::string::npos)
        {
            count += 1;
        }
    }
    return count;
}

File: Beginner/2_Count_Vowels/tests/count_vowels.py
import unittest
import random


class TestCountVowels(unittest.TestCase):

    @classmethod
    def setUpClass(cls) -> None:
        print("Testing Count Vowels in a string with CPython")
        return super().setUpClass()

    @staticmethod
    def get_random_string() -> str:
        alphabet = "abcdefghijklmnopqrstuvwxyz"
        return "".join([
            random.choice(alphabet)
            for i in range(50)
        ])

    @staticmethod
    def count_vowels(random_string: str) -> int:
        vowels_list = "aeiou"
        return len([
            None
            for letter in random_string
            if letter in vowels_list
        ])

    def test_count_vowels(self):
        random_string = self.get_random_string()
        expected = self.count_vowels(random_string)
        try:
            from CPythonLearn2 import count_vowels
            self.assertEqual(
                expected,
                count_vowels(random_string),
                f'The vowels count should be : {expected}'
            )
        except ImportError as e:
            self.assertTrue(
                False,
                f'The importation failed due to this error: {e}'
            )
